stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/DisconnectCause.kt

```diff
@@ -0,0 +1,5 @@
+package io.getstream.chat.android.client.clientstate
+
+public enum class DisconnectCause {
```

Comment:
Could it be a Sealed Class instead?
The "Error" case could have some "usefull info" about what happens, instead of only "error"

Reply:
What kind of info? ChatError? Where we can get it? We use it when `ChatSocketServiceImpl` we observe state transformation to `State.DisconnectedTemporarily`

Reply:
I'm not sure if we could know "why" the error happend, but, keeping it as a Sealed Class, we are able to extend it easilly later. For now, if we don't have extra info for that, we can use only `object` instances